code/backtracking/src/sudoku_solve/sudoku_solveNxN.cpp

```diff
@@ -0,0 +1,96 @@
+
+#include<bits/stdc++.h>
+
+using namespace std;
+
+const int n = 9;
+
+int grid[n][n];
+
+void input_grid() {
+    int i, j;
+    for (i = 0; i < n; i++) {
+        for (j = 0; j < n; j++) {
+            cin >> grid[i][j];
+        }
+    }
+}
+
+void print_grid() {
+    int i, j;
+    for (i = 0; i < n; i++) {
+        for (j = 0; j < n; j++) {
+            cout << grid[i][j] << " ";
+        }
+        cout << '\n';
+    }
+}
+
+bool can_be_placed(int row, int col, int num) {
+    int ii, jj;
+    ii = 0, jj = col;
+    // row check
+    while (ii < n) {
+        if (grid[ii][jj] == num) {
+            return false;
+        }
+        ii++;
+    }
+    // col check
+    ii = row; jj = 0;
+    while (jj < n) {
+        if (grid[ii][jj] == num) {
+            return false;
+        }
+        jj++;
+    }
+    // 3*3 check
+    int startrow, startcol;
+    startrow = (row / 3) * 3;
+    startcol = (col / 3) * 3;
+    for (ii = startrow; ii < startrow + 3; ii++) {
+        for (jj = startcol; jj < startcol + 3; jj++) {
+            if (grid[ii][jj] == num) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+void sudoku_solver(int row, int col) {
+    // cout<<row<<" "<<col<<'\n';
```

Comment:
Remove this.

Reply:
It's for user so they can comment it out and check for there case for more clarity